demos/rag_agentic/src/agent_with_mcp_ocp_slack.py

```diff
@@ -0,0 +1,110 @@
+        provider_id="model-context-protocol",
+        mcp_endpoint={"uri":mcp_url},
+        )
+mcp_tools = [t.identifier for t in client.tools.list(toolgroup_id="mcp::slack")]
+
+logger.info(f"""Your Server has access the the following toolgroups:
+{set(registered_toolgroups)}
+""")
+
+# system prompts for different models
+llama_prompt = """You are a helpful assistant. You have access to a number of tools.
+    Whenever a tool is called, be sure return the Response in a friendly and helpful tone."""
+granite_prompt = """You are a helpful AI assistant with access to the tools listed next. When a tool is required to answer the user's query, respond with `<tool_call>` followed by a JSON object of the tool used. For example: `<tool_call> {"name":"function_name","arguments":{"arg1":"value"}} </tool_call>`:The user will respond with the output of the tool execution response so you can continue with the rest of the initial user prompt (continue).
+If a tool does not exist in the provided list of tools, notify the user that you do not have the ability to fulfill the request."""
+
+# Create simple agent with tools
+agent = Agent(
+    client,
+    model=llama_model,
+    instructions = llama_prompt,
+    tools=["mcp::slack", "mcp::openshift"],
+    tool_config={"tool_choice":"auto"},
+    sampling_params={"max_tokens":4096, "strategy":{"type": "greedy"},}
+)
+
+if args.auto:
+    user_prompts = ["""View the logs for pod hello-world in the llama-serve OpenShift namespace. Categorize it as normal or error.""",
+                    """Summarize the results with the pod name, category along with a briefly explaination as to why you categorized it as normal or error. Respond with plain text only. Do not wrap your response in additional quotation marks.""",
+                    """Send a message with the summarization to the demos channel on Slack."""]
+    session_id = agent.create_session(session_name="Auto_demo")
+    for i, prompt in enumerate(user_prompts):
+        turn_response = agent.create_turn(
+            messages=[
+                {
+                    "role":"user",
+                    "content": prompt
+                }
+            ],
+            session_id=session_id,
+            stream=True,
+        )
+        for log in EventLogger().log(turn_response):
+            log.print()
+
+else:
+    #Create a chat session
+    session_id = agent.create_session(session_name="Conversation_demo")
+    while True:
+        user_input = input(">>> ")
+        if "/bye" in user_input:
+            if args.session_info_on_exit:
+                agent_session = client.agents.session.retrieve(session_id=session_id, agent_id=agent.agent_id)
+                print( agent_session.to_dict())
+            break
+        turn_response = agent.create_turn(
+            session_id=session_id,
+            messages=[{"role": "user", "content": user_input}],
+        )
+
+        for log in EventLogger().log(turn_response):
+            log.print()
```

Comment:
Is this section necessary at all? If we just want to test out this prompt chain you can probably take this out, as well as the `--auto` argument.

Reply:
sure 👍